### PR TITLE
remove all instances of boost foreach, replace with c++11 loops

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

**Review by awemany — APPROVED** [review]
Despite the travis failing, this all looks good to me. Didn't check out and compile, though. Did you use/write some script to auto-convert this?

**Comment by ptschip** [comment]
never mind removing the prevector, i imagine it's related to this PR, in that you need it for the reverse iterator.

File: src/httprpc.cpp [hunk 1]
```diff
@@ -90,7 +89,7 @@ static bool multiUserAuthorized(std::string strUserPass)
     if (mapMultiArgs.count("-rpcauth") > 0)
     {
         // Search for multi-user login/pass "rpcauth" from config
-        BOOST_FOREACH (std::string strRPCAuth, mapMultiArgs["-rpcauth"])
+        for (std::string strRPCAuth : mapMultiArgs["-rpcauth"])
         {
```
Comment (ptschip) [inline src/httprpc.cpp]: minor, but couldn't we do

- for (std::string strPRCAuth...
+for (std::string &strRPCAuth...

File: src/init.cpp [hunk 2]
```diff
@@ -1271,7 +1271,7 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
 
     // sanitize comments per BIP-0014, format user agent and check total size
     std::vector<string> uacomments;
-    BOOST_FOREACH (string cmt, mapMultiArgs["-uacomment"])
+    for (string cmt : mapMultiArgs["-uacomment"])
     {
```
Comment (ptschip) [inline src/init.cpp]: same here,  use 

+for (string &cmt

**Comment by Greg-Griffith** [comment]
@ptschip yes, i did need it specifically for the reverse iterator. so ok, ill leave it in this pr instead of picking it out into another. 
As for the other comments, i didn't change anything about the loops except remove the use of boost and replace it with c++11 for. Do you want me to go through the loops and re-evaluate them as well?

File: src/rpc/server.cpp [hunk 3]
```diff
@@ -64,7 +63,7 @@ void RPCServer::OnPostCommand(boost::function<void(const CRPCCommand &)> slot)
 void RPCTypeCheck(const UniValue &params, const list<UniValue::VType> &typesExpected, bool fAllowNull)
 {
     unsigned int i = 0;
-    BOOST_FOREACH (UniValue::VType t, typesExpected)
+    for (UniValue::VType t : typesExpected)
     {
```
Comment (ptschip) [inline src/rpc/server.cpp]: can can we add a reference here?

+for(UniValue::VType &t : ....

**Comment by gandrewstone** [comment]
run clang formatting...

**Comment by ptschip** [comment]
i have been fixing them every time I replaced with c++11 loops...there
were occasional places where a reference was not being used when one
could be...we should fix those when we come across them...here in you're
PR It hink there were 2 ot 3 intances that could be fixed.

**Review by ptschip — APPROVED** [review]

**Review by sickpig — APPROVED** [review]

**Pull request merged**